# The customized R-scripts for the 1000 Soils metagenome analyses #

The scripts in this folder were implemented to generate visualizations featured in the manuscript, "Continental-scale integration of soil metagenomes and metabolomes reveals ubiquitous microbial capacity for recalcitrant carbon decomposition.‚Äù The only exception to this description are the codes contained in the folder, **Figure_3_and_Suppl_Fig_2**, which contains codes used to conduct the Spearman Correlation alaysis between the metagenome and the FTICR data. Below are the general specifications used to test the codes, and brief description of the codes in each of the subfolders. 

## Requirments ##

- **OS**: All codes were run in MacOS Sequoia (15.7.3)
- **Processor and RAM**: Apple M2 Pro; 16GB RAM
- **R version**: R version 4.5.1 (2025-06-13)
- **Software used**: R Studio version 2024.12.1+563

All codes (Except Figure 3 and Supplementary Figure 2) was tested using the settings listed above and was completed within 3 minutes. The Spearman Correlation analyses implemented as part of the codes located in the **Figure_3_and_Suppl_Fig_2** folder was completed within 30 minutes.

## Instructions ##

Download the codes from this repository and place them in a way such that the R code and the **Sample_Data** folder are in the same path. Please refer to the README for each codes to see the list of R libraries used.

## Description of the codes and the sample data ##

- **Figure_1**: The code in this folder was used to generate the heatmap and the bar charts depicting the microbial diversity of the soils. The heatmap data was generated by using the customized Python code to compile the counts of the reads mapped to the dereplicated MAGs into a genus level. The bar chart data was generated by compiling the GTDBTk assignment of the MAGs into raw counts of the bacterial and archaeal genus detected in the soil.
- **Figure_2**: The code in this folder is a modified version of the R package, *ftmsRanalysis* (Degnan et al., 2025; https://analyticalsciencejournals.onlinelibrary.wiley.com/doi/10.1002/rcm.9980). The modification was made to the existing code to generate stacked bar charts representing distribution of SOM classes that are unique to the soil types under investigation.
- **Figure_3_and_Suppl_Fig_2**: The codes in this figure was used to conduct the Spearman correlation analysis between the abundances of the KOs detected in the MAGs and the SOM measurements across the samples. The resulting matrices were filtered using the parameters described in the **Methods** section in the manuscript, and were imported to Cytoscape to generate the network diagram shown in the manuscript.
- **Figure_4**: The code generates a bubble chart that represents the distribution of the KEGG pathways based on recruitment of KO-annotated genes mapped to the FTICR-MS derived compounds. The summary table of the KOs mapped to the FTICR-derived molecules was generated using the *ftmsRanalysis* package. The table containing the ratio of KOs affiliated with specific pathways versus the total number of KOs in each soils was generated by using the *enrichKEGG* algorithm in the clusterProfiler R package (Honeker et al., 2023; https://www.nature.com/articles/s41564-023-01432-9).
- **Figure_5_and_Suppl_Fig_3**: The codes in this folder were used to generate an upset plot and a stacked bar chart representing the composition of the CAZyme families present in the archaeal and bacterial MAGs detected in the soils (Figure 5) and in the fungal contigs (Supplementary Figure 3). The output summary tables of the *dbCAN2* (Zhang et al., 2018; https://academic.oup.com/nar/article/46/W1/W95/4996582) were compiled into a single tab-separated file using a customized Python script.
- **Figure_6**: The code in this folder was used to generate a bar-and-whisker plot representing the uptake rates of the compounds calculated using the KBase narrative, *MONet_1000_Soils_metaG_path_model_w_auto_import* (https://narrative.kbase.us/narrative/229872). A customized Python script was used to extract the uptake rate field from the tables generated from the KBase narrative, and compile them into a single table that was applied in the R code.
