# The library specifications of the R codes  #

Please refer to the list below for the library packages required for each codes contained in this folder.


## Figure_1 ##

The code in this folder was used to generate the heatmap and the bar charts depicting the microbial diversity of the soils. The heatmap data was generated by using the customized Python code to compile the counts of the reads mapped to the dereplicated MAGs into a genus level. The bar chart data was generated by compiling the GTDBTk assignment of the MAGs into raw counts of the bacterial and archaeal genus detected in the soil.

### Required data (See Sample_Data) ###
- **1000_Soils_drep_MAGs_row_col_cmp.csv**: A comma-separated file (CSV) containing the counts of genus-level taxa across the soil types. The rows have the GTDB classifications of the genera, while the columns represent the types of the soils, classified based on depth and respiration level.
- **1000_Soils_drep_MAGs_classify_genus_ct.csv**: A CSV file containing the counts of the MAGs assigned to the genus-level taxonomy using GTDB. The first column shows the names of the genera, while the second column shows the counts of each genus.

### Required R library packages ###
- RColorBrewer
- ggplot2
- reshape
- rstudioapi

## Figure_2 ##

The code in this folder is a modified version of the R package, *ftmsRanalysis* (Degnan et al., 2025; https://analyticalsciencejournals.onlinelibrary.wiley.com/doi/10.1002/rcm.9980). The modification was made to the existing code to generate stacked bar charts representing distribution of SOM classes that are unique to the soil types under investigation.

### Required data (See Sample_Data) ###
- **1000_Soil_FTICR_all_samples_mass.csv**: A CSV file containing the raw counts of the compounds (rows) detected across the samples (columns). This file was derived from the FTICR experiments. The masses shown in the first column are specific to the compounds.
- **fdata_1000soils.csv**: A metadata of the samples. Contains information regarding the soil depths and respiration level of each sample.

### Required R library packages ###
**Note:** refer to the instructions in lines 23-31 for installing hexbin, devtools, datadr and reticulate. These are packages for connecting the Python dependencies required to draw the images in SVG format.
- here
- ftmsRanalysis
- dplyr
- ggplot2
- plotly
- hexbin
- devtools
- reticulate
- datadr

## Figure_3_and_Suppl_Fig_2 ##

The codes in this figure was used to conduct the Spearman correlation analysis between the abundances of the KOs detected in the MAGs and the SOM measurements across the samples. The resulting matrices were filtered using the parameters described in the **Methods** section in the manuscript, and were imported to Cytoscape to generate the network diagram shown in the manuscript.

### Required data (See Sample_Data) ###
- **A tab-separated table representing the abundance of the KOs detected from the MAGs (e.g., 1000_Soils_drepd_subsoil_hi_KO.tsv)**: The output of EnrichM (see the **Methods** section) were processed using a customized Python script to generate these tables.
- **A CSV table containing the abundances of the SOM compounds across the samples (e.g., 1000_Soils_FTICR_subsoil_hi_filtered.csv)**: The raw output of the FTICR experiments were processed using a customized Python script to generate these tables.

### Required R library packages ###
- Hmisc
- dplyr
- tidyverse
- NOISeq
- reshape2
- rstudioapi

## Figure_4 ## 

The code generates a bubble chart that represents the distribution of the KEGG pathways based on recruitment of KO-annotated genes mapped to the FTICR-MS derived compounds. The summary table of the KOs mapped to the FTICR-derived molecules was generated using the *ftmsRanalysis* package. The table containing the ratio of KOs affiliated with specific pathways versus the total number of KOs in each soils was generated by using the *enrichKEGG* algorithm in the clusterProfiler R package (Honeker et al., 2023; https://www.nature.com/articles/s41564-023-01432-9).

### Required data (See Sample_Data) ###
- **FTICR_KO_50_pct_num_KOs.csv**: A CSV file containing the number of KOs (i.e., number of functional genes) detected in each of the four soil types. Used to generate the bar chart. This table was generated by summing the raw counts of KOs derived from *EnrichM*.
- **FTICR_KO_50_oct_top_20.tsv**: A tab-separated file generated from the *ftmsRanalysis* package.

### Required R library packages ###
- RColorBrewer
- ggplot2
- reshape
- dplyr
- rstudioapi

## Figure_5_and_Suppl_Fig_3 ##

The codes in this folder were used to generate an upset plot and a stacked bar chart representing the composition of the CAZyme families present in the archaeal and bacterial MAGs detected in the soils (Figure 5) and in the fungal contigs (Supplementary Figure 3). The output summary tables of the *dbCAN2* (Zhang et al., 2018; https://academic.oup.com/nar/article/46/W1/W95/4996582) were compiled into a single tab-separated file using a customized Python script.

### Required data (See Sample_Data) ###

These files are required for the code *Fig_5_upset_genus.R*:

- **1000_Soils_drep_MAGs_dbCAN_avg_genus.csv**: A CSV file the raw counts of CAZyme families detected across the soil types.  Customized Python scripts were used to extracti a set of CAZyme families that were detected in the archaeal and bacterial MAGs using two or more of the search methods present in the *dbCAN2*. The counts of these CAZyme families were then compiled together to generate this input CSV file.
- **fungal_50pct_CAZy.csv**: Similar to the file described above, but for fungal contigs.
- **metadata.csv**: A CSV file required for indexing purpose (used in the *upset* method).

The following is required for the code *Fig_5_upset_genus_CAZyme_distr_stacked_bar.R*:

- **1000_Soils_drep_MAGs_CAZy_cat_assign.tsv**: An upset plot is essentially a Venn diagram generated from overlapping different data sets. This overlap generates separate segments (e.g., unique in group A, present in all groups, etc). This table contains the assignments of the CAZyme families into one of these segments. 

### Required R library packages ###
- ggplot2
- dplyr
- rstudioapi

## Figure_6 ##

The code in this folder was used to generate a bar-and-whisker plot representing the uptake rates of the compounds calculated using the KBase narrative, *MONet_1000_Soils_metaG_path_model_w_auto_import* (https://narrative.kbase.us/narrative/229872). A customized Python script was used to extract the uptake rate field from the tables generated from the KBase narrative, and compile them into a single table that was applied in the R code.

## Supple_Fig_1 ##
The code in this folder parses a table containing geographical coordinates of the sample locations and generates a map of these sites.
